gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/DiskStoreImpl.java

```diff
@@ -388,7 +388,17 @@
   private DiskStoreID diskStoreID;
 
   private volatile Future lastDelayedWrite;
-  
+
+  /**
+   * Currently active disk block sorter used by region iterators for
+   * cross iterator sorting in case multiple concurrent iterators are open.
+   * The iterators will grab hold of currently active sorter and submit their
+   * disk blocks to be sorted and at some point one of those iterators will
+   * open the sorter for reading at which point no new blocks can be added
+   * and a new current sorter will be created.
+   */
+  private final DiskBlockSortManager sortManager;
```

Comment:
Does closing of DiskStoreImpl ( dropping of DiskStore ) should clean the current active sorter in some special way or GC is good enough?

Reply:
No special cleanup is required. The iteration checks for cancelInProgress for this reason to terminate quickly.